### ISSUE TITLE
How scoring is done

### ISSUE THREAD

**Issue by GitToby** [liufan2312/tgs-salt-identification#2]
This competition is evaluated on the mean average precision at different intersection over union (IoU) thresholds. The IoU of a proposed set of object pixels and a set of true object pixels is calculated as:

IoU(A,B)=A∩B/A∪B. (Jaccard Sim over the two sets)

The metric sweeps over a range of IoU thresholds, at each point calculating an average precision value. The threshold values range from 0.5 to 0.95 with a step size of 0.05: (0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95). In other words, at a threshold of 0.5, a predicted object is considered a "hit" if its intersection over union with a ground truth object is greater than 0.5.

At each threshold value t
, a precision value is calculated based on the number of true positives (TP), false negatives (FN), and false positives (FP) resulting from comparing the predicted object to all ground truth objects:

TP(t)/(TP(t)+FP(t)+FN(t))

A true positive is counted when a single predicted object matches a ground truth object with an IoU above the threshold. A false positive indicates a predicted object had no associated ground truth object. A false negative indicates a ground truth object had no associated predicted object. The average precision of a single image is then calculated as the mean of the above precision values at each IoU threshold:

1/|thresholds| * ∑_{t}TP(t)/(TP(t)+FP(t)+FN(t))

Lastly, the score returned by the competition metric is the mean taken over the individual average precisions of each image in the test dataset.
Submission File

In order to reduce the submission file size, our metric uses run-length encoding on the pixel values. Instead of submitting an exhaustive list of indices for your segmentation, you will submit pairs of values that contain a start position and a run length. E.g. '1 3' implies starting at pixel 1 and running a total of 3 pixels (1,2,3).

The competition format requires a space delimited list of pairs. For example, '1 3 10 5' implies pixels 1,2,3,10,11,12,13,14 are to be included in the mask. The pixels are one-indexed and numbered from top to bottom, then left to right: 1 is pixel (1,1), 2 is pixel (2,1), etc.

The metric checks that the pairs are sorted, positive, and the decoded pixel values are not duplicated. It also checks that no two predicted masks for the same image are overlapping.

The file should contain a header and have the following format. Each row in your submission represents a single predicted salt segmentation for the given image.
```
id,rle_mask
3e06571ef3,1 1
a51b08d882,1 1
c32590b06f,1 1
etc.
```

**Comment by liufan2312** [comment]
why do they only give us 4000 training images while they have 18000 testing images lol

**Comment by liufan2312** [comment]
# Weekly catch-up talking points
### To-do:
- Change image characteristics such as contrast
- Use pre-trained network to initialize

### In progress:
- Feature engineering such as edge detection **

### Done:
- Add batch normalization ***
- More data augmentation: elastic transformation ***
- Use small feature maps to test the changes ****
- Change batch size and test performance ****
